closure/goog/net/websocket.js

```diff
@@ -62,7 +62,7 @@ goog.require('goog.log');
  */
 goog.net.WebSocket = function(opt_params, opt_getNextReconnect) {
   goog.net.WebSocket.base(this, 'constructor');
-  if (typeof opt_params != 'object') {
+  if (typeof opt_params != 'object' || opt_params === null) {
     opt_params = /**@type {!goog.net.WebSocket.Options} */ ({
       autoReconnect: opt_params,
```

Comment:
This assignment isn't safe when `opt_params === null`.

Reply:
Actually I realized that this is OK (but doesn't look great) because the next line forces `this.autoReconnect_` to be boolean. That being said, is it still possible to do what I said in the top-level comment, because technically the type annotations should guard against this already?

Reply:
Yes, it does slightly differ from convention to do so but I can make a change to avoid passing null here. However, I do think adding a null check here (and really anywhere where typeof "object" comparison is done) makes the intent clearer for additions of future logic.

Either way, I don't feel very strongly about this so feel free to close the issue if you disagree.

Reply:
Actually, we are thinking about changing this to only allowing the object parameter, which should simplify this a bit. In this case we just check if the input parameter is falsey or not.

Reply:
Makes sense!